### PR TITLE
Separate core into a standalone crate

### PR DESCRIPTION
**Pull request by mirgee**
Relocates `BaseLedger`,` BaseAnonCreds`, `BaseWallet` trait definitions and their corresponding implementations (including the `indy` module and related `utils`) into a distinct crate called `aries_vcx_core`. As a result, `aries-vcx` now depends on `indy-vdr`, `indy-credx`, and  `libvdrtools` indirectly through this new crate, which doesn't reexport any of them.

The primary objective of this pull request is the separation of these components, and no additional efforts were made to enhance the codebase beyond this specific aim.

### CODE REVIEW

**Review by Patrik-Stas — DISMISSED** [review]

File: aries_vcx/src/errors/mapping_others.rs [hunk 1]
```diff
@@ -21,3 +23,145 @@ impl<T> From<PoisonError<T>> for AriesVcxError {
...
+            AriesVcxErrorKind::InvalidHttpResponse => AriesVcxCoreErrorKind::InvalidHttpResponse,
+            AriesVcxErrorKind::InvalidMessages => AriesVcxCoreErrorKind::InvalidMessages,
+            AriesVcxErrorKind::VdrToolsError(u32) => AriesVcxCoreErrorKind::VdrToolsError(u32),
+            AriesVcxErrorKind::UrsaError => AriesVcxCoreErrorKind::UrsaError,
+            AriesVcxErrorKind::NoAgentInformation => AriesVcxCoreErrorKind::NoAgentInformation,
+            AriesVcxErrorKind::InvalidMessageFormat => AriesVcxCoreErrorKind::InvalidMessageFormat,
+        };
+        AriesVcxCoreError::from_msg(kind, format!("AriesVcxError: {}", err.to_string()))
+    }
+}
```
Comment (bobozaur) [inline aries_vcx/src/errors/mapping_others.rs:167]: I bet you were annoyed having to do this 😆 .

We can do better in terms of error handling, but it's definitely outside the scope of this PR as it affects the entire workspace.

**Review by bobozaur — APPROVED** [review]
I only left a comment related to error handling, which is not meant for this PR per se but more as a note to all of us.

Other than that, LGTM.

**Review by gmulhearn — DISMISSED** [review]
I probably won't have time to give this an in-depth, but I definitely approve of the idea. Good improvement for vcx. Will put my review as approval assuming others have had a deeper look at the implementation

**Review by gmulhearn — COMMENTED** [review]
see my review above ^. But i would like to see `skip-ci` removed before any merging. just as a confirmation

**Comment by Patrik-Stas** [comment]
@gmulhearn good catch! Rerunning

**Review by Patrik-Stas — APPROVED** [review]

**Comment by Patrik-Stas** [comment]
Running full CI revealed few issues (compile and CI issues), go it fixed, not a big changes so I am merging per previous approvals.

**Pull request merged**